tests/test_async_uploader.py

```diff
@@ -54,6 +55,34 @@ def test_upload(self):
             self.assertEqual(self.async_uploader.offset,
                              self.async_uploader.get_file_size())
 
+    def test_upload_empty(self):
+        with aioresponses() as resps:
+            resps.post(
+                "http://tusd.tusdemo.net/files/", status=200,
+                headers={
+                    "upload-offset": "0",
+                    "location": "http://tusd.tusdemo.net/files/this-is-not-used"
+                }
+            )
+            resps.patch(
+                "http://tusd.tusdemo.net/files/this-is-not-used",
+                exception=ValueError(
+                    "PATCH request not allowed for empty file"
+                )
+            )
+
+            # Set uploader's URL to None. This corresponds to a new upload;
+            # if the URL is set we're continuing an existing upload.
+            self.async_uploader.url = None
```

Comment:
Is it necessary to set `url = None` here? I thought we are creating a new uploader for each test, so the url should not be initialized here. Or am I missing something?

Reply:
It's not actually necessary, and it's a better approach to create an uploader from the existing `self.client` fixture. I've refactored the tests to do so.